code/Model/SiteTree.php

```diff
@@ -1604,6 +1604,21 @@ public function generateURLSegment($title)
         // Hook for extensions
         $this->extend('updateURLSegment', $t, $title);
 
+        $excludes = Director::config()->get('rules');
+        $excludes = array_keys($excludes);
+        $disallowed = array_map(function ($key) {
+            $route = explode('/', $key);
+            return $route[0];
```

Comment:
Could you do the `$` check in here, and just put array_filter around the entire array_map? It would clean up the complexity of the code here a bit.